crates/biome_css_analyze/src/lint/nursery/no_unknown_unit.rs

```diff
@@ -0,0 +1,195 @@
+use biome_analyze::{context::RuleContext, declare_rule, Ast, Rule, RuleDiagnostic, RuleSource};
+use biome_console::markup;
+use biome_css_syntax::{
+    AnyCssDimension, CssFunction, CssGenericProperty, CssQueryFeaturePlain, CssSyntaxKind,
+};
+use biome_rowan::{SyntaxNodeCast, TextRange};
+
+const RESOLUTION_MEDIA_FEATURE_NAMES: [&str; 3] =
+    ["resolution", "min-resolution", "max-resolution"];
+
+// Check if the value is a CSS hack used in Internet Explorer.
+fn is_css_hack(value: &str) -> bool {
+    value == "\\0"
+}
+
+declare_rule! {
+    /// Disallow unknown CSS units.
+    ///
+    /// For details on known CSS units, see the [MDN web docs](https://developer.mozilla.org/en-US/docs/Learn/CSS/Building_blocks/Values_and_units#lengths).
+    ///
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```css,expect_diagnostic
+    /// a {
+    ///   width: 10pixels;
+    /// }
+    /// ```
+    ///
+    /// ```css,expect_diagnostic
+    /// a {
+    ///   width: calc(10px + 10pixels);
+    /// }
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```css
+    /// a {
+    ///   width: 10px;
+    /// }
+    /// ```
+    ///
+    /// ```css
+    /// a {
+    ///   width: 10Px;
+    /// }
+    /// ```
+    ///
+    /// ```css
+    /// a {
+    ///   width: 10pX;
+    /// }
+    /// ```
+    ///
+    /// ```css
+    /// a {
+    ///   width: calc(10px + 10px);
+    /// }
+    /// ```
+    ///
+    pub NoUnknownUnit {
+        version: "next",
+        name: "noUnknownUnit",
+        recommended: true,
+        sources: &[RuleSource::Stylelint("unit-no-unknown")],
+    }
+}
+
+pub struct NoUnknownUnitState {
+    unit: String,
+    span: TextRange,
+}
+
+impl Rule for NoUnknownUnit {
+    type Query = Ast<AnyCssDimension>;
+    type State = NoUnknownUnitState;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Option<Self::State> {
+        let node = ctx.query();
+
+        match node {
+            AnyCssDimension::CssUnknownDimension(dimension) => {
+                let unit_token = dimension.unit_token().ok()?;
+                let unit = unit_token.text_trimmed().to_string();
+
+                // Ignore CSS hack because it's parsed as an unknown unit.
+                if is_css_hack(&unit) {
+                    return None;
+                }
+
+                Some(NoUnknownUnitState {
+                    unit,
+                    span: unit_token.text_trimmed_range(),
+                })
+            }
+            AnyCssDimension::CssRegularDimension(dimension) => {
+                let unit_token = dimension.unit_token().ok()?;
+                let unit = unit_token.text_trimmed().to_string();
+
+                if unit == "x" {
```

Comment:
Can you add a comment that explains this "X" check. You will provide context to other developers

Reply:
I have added the explanation as a comment.